src/clarity-angular/emphasis/alert/alert.ts

```diff
@@ -10,20 +10,41 @@ import {
     Output
 } from "@angular/core";
 
+// providers
+import { AlertIconAndTypesService } from "./providers/icon-and-types-service";
+
 @Component({
     selector: "clr-alert",
+    providers: [ AlertIconAndTypesService ],
     templateUrl: "./alert.html"
 })
 export class Alert {
+    constructor(public iconService: AlertIconAndTypesService) {
+    }
+
     @Input("clrAlertSizeSmall") isSmall: boolean = false;
-    @Input("clrAlertType") type: string = "alert-info";
     @Input("clrAlertClosable") closable: boolean = true;
     @Input("clrAlertAppLevel") isAppLevel: boolean = false;
 
     @Input("clrAlertClosed") _closed: boolean = false;
     @Output("clrAlertClosedChange") _closedChanged: EventEmitter<boolean> = new EventEmitter<boolean>(false);
 
-    private alertTypes: string[] = [ "alert-info", "alert-warning", "alert-danger", "alert-success"];
+    @Input("clrAlertType")
+    set alertType(val: string) {
+        this.iconService.alertType = val;
+    }
+    get alertType(): string {
+        return this.iconService.alertType;
+    }
+
+    @Input("clrAlertIcon")
+    set alertIconShape(value: string) {
+        this.iconService.alertIconShape = value;
+    }
```

Comment:
I will have to think about this one. Mainly, I did this so that this would not be a breaking change for people using `clr-alert` today. If we want people to project anything into the icon wrapper, that's a little different from what clarity alerts have always done. At least now, they can use any custom icon they want.

I would say that this suggestion would require a templateRef in the DOM and potentially a fallback to clarity icons. I don't know if that's something we can manage in this changeset.